### PR TITLE
Jenkins: add jobs for GKE+Trusty

### PR DESCRIPTION
**Pull request by wonderfly**
Add three jobs to run e2e tests against GKE prod/staging/test API endpoints with
Trusty as node image.

@roberthbailey @spxtr 

cc/ @andyzheng0831

### CODE REVIEW

**Comment by wonderfly** [comment]
I guess it still needs a LGTM.

**Comment by roberthbailey** [comment]
Yes. lgtm and ok-to-merge serve different purposes.

**Comment by roberthbailey** [comment]
@ixdy (your reviewer) should apply the lgtm label.

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -806,6 +806,59 @@ case ${JOB_NAME} in
...
+
+  kubernetes-e2e-gke-trusty-staging)
+    # Override GKE default to use rc Cloud SDK and staging endpoint
+    CLOUDSDK_BUCKET="gs://cloud-sdk-build/testing/rc"
+    GKE_API_ENDPOINT="https://staging-container.sandbox.googleapis.com/"
+    : ${E2E_CLUSTER_NAME:="jkns-gke-e2e-staging-trusty"}
+    : ${E2E_NETWORK:="e2e-gke-trusty-staging"}
+    : ${E2E_SET_CLUSTER_API_VERSION:=y}
+    : ${JENKINS_USE_SERVER_VERSION:=y}
+    : ${PROJECT:="e2e-gke-trusty-staging"}
```
Comment (ixdy) [inline hack/jenkins/e2e.sh]: is this correct? it's using a different format from the other two jobs.

Comment (wonderfly) [inline hack/jenkins/e2e.sh]: You mean the `PROJECT` name? If so, it's intended. GCE has a limit of 30 chars on project names, and `kubekins-gke-e2e-trusty-staging` is over the limit so I decided to trim it. You are right it's not analogical to others, but I don't think it matters, and this is the project that I have credentials, quotas and other acls set up.

Comment (ixdy) [inline hack/jenkins/e2e.sh]: ok, just wanted to make sure that it was intentional.

**Comment by ixdy** [comment]
LGTM

**Comment by spxtr** [comment]
LGTM

The jobs are already up and running on Jenkins, and failing because this isn't in yet. Probably worth manually merging since this doesn't get hit by CI.

**Pull request merged**